lib/google/gax/errors.rb

```diff
@@ -27,20 +27,26 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+require 'English'
+
 module Google
   module Gax
     # Common base class for exceptions raised by GAX.
     class GaxError < StandardError
-      attr_reader :cause
-
       # @param msg [String] describes the error that occurred.
-      # @param cause [Error] the exception raised by a lower layer of
-      #   the RPC stack (for example, gRPC) that caused this
-      #   exception, or None if this exception originated in GAX.
-      def initialize(msg, cause: nil)
-        msg = "GaxError #{msg}, caused by #{cause}" if cause
+      def initialize(msg)
+        msg = "GaxError #{msg}"
+        msg += ", caused by #{$ERROR_INFO}" if $ERROR_INFO
         super(msg)
-        @cause = cause
+        @cause = $ERROR_INFO
+      end
+
+      # cause is a new method introduced in 2.1.0, bring this
+      # method if it does not exist.
+      unless respond_to?(:cause)
```

Comment:
Just to check -- what versions of Ruby are we committed to supporting? It doesn't look like our CI checks with older versions of Ruby, so such code is non-testable at the moment.

If we want to support older versions of Ruby, then we should be testing them with Travis; if we don't want to support them, then we don't need this code.